src/sinks/new_relic/model.rs

```diff
@@ -57,39 +41,70 @@ impl TryFrom<Vec<Event>> for MetricsApiModel {
 
         for buf_event in buf_events {
             if let Event::Metric(metric) = buf_event {
-                // Future improvement: put metric type. If type = count, NR metric model requires an interval.ms field, that is not provided by the Vector Metric model.
-                match metric.value() {
-                    MetricValue::Gauge { value } => {
-                        metric_array.push((
-                            Value::from(metric.name().to_owned()),
-                            Value::from(
-                                NotNan::new(*value).map_err(|_| {
-                                    NewRelicSinkError::new("NaN value not supported")
-                                })?,
-                            ),
-                            Value::from(metric.timestamp()),
-                        ));
+                // Generate Value::Object() from BTreeMap<String, String>
+                let (series, data, _) = metric.into_parts();
+                let attr = if let Some(tags) = series.tags {
+                    let mut bt = BTreeMap::new();
+                    for (key, value) in tags {
+                        bt.insert(key, Value::from(value));
+                    }
+                    Some(Value::from(bt))
+                } else {
+                    None
+                };
+
+                let mut metric_data = KeyValData::new();
+
+                if let MetricValue::Gauge { value } | MetricValue::Counter { value } = data.value {
+                    metric_data.insert("name".to_owned(), Value::from(series.name.name));
+                    metric_data.insert(
+                        "value".to_owned(),
+                        Value::from(
+                            NotNan::new(value)
+                                .map_err(|_| NewRelicSinkError::new("NaN value not supported"))?,
+                        ),
+                    );
+                    metric_data.insert(
+                        "timestamp".to_owned(),
+                        if let Some(ts) = data.time.timestamp {
+                            Value::from(ts.timestamp())
+                        } else {
+                            Value::from(DateTime::<Utc>::from(SystemTime::now()).timestamp())
+                        },
```

Comment:
```suggestion
                        Value::from(data
                            .time
                            .timestamp
                            .unwrap_or_else(|| DateTime::<Utc>::from(SystemTime::now()))
                            .timestamp()),
```

I think that inner `DateTime` line might just be `Utc::now()` too.